packages/beacon-node/src/network/reqresp/handlers/dataColumnSidecarsByRange.ts

```diff
@@ -31,7 +34,10 @@ export async function* onDataColumnSidecarsByRange(
     for (let slot = startSlot; slot < endSlot; slot++) {
       const dataColumnSidecars = await finalized.getManyBinary(slot, availableColumns);
 
-      for (const dataColumnSidecarBytes of dataColumnSidecars) {
+      const unavailableColumnIndices: ColumnIndex[] = [];
+      for (let i = 0; i < dataColumnSidecars.length; i++) {
+        const dataColumnSidecarBytes = dataColumnSidecars[i];
+        // for (const dataColumnSidecarBytes of dataColumnSidecars) {
```

Comment:
The function `validateRequestedDataColumns` already have `missingColumns` and it logs that as well. We should just return that from the function and report to metrics and don't change any other logic.

Reply:
thats columns that were requested but we do not expect to find in the database.  what i added was for columns that we do expect to have but do not (error in our implementation)